tests/unit/test.config.js

```diff
@@ -259,6 +270,43 @@ describe('config', () => {
       const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
       assert.strictEqual(newArgv.sourceDir, '/configured/directory');
     });
+
+    it('throws an error when an option is not camel cased', () => {
+      const {argv, defaultValues} = makeArgv({
+        globalOpt: {
+          'source-dir': {
+            type: 'string',
+            demand: false,
+          },
+        },
+      });
+      const configObject = {
+        'source-dir': 'fake/value/',
+      };
+      assert.throws(() => {
+        applyConfigToArgv({argv, configObject, defaultValues});
```

Comment:
you could make a shorter helper function just for testing to make it easier to omit defaults. Something like this:
````js
const applyConf = (params) => applyConfigToArgv({
  configFilename: 'some/path/to/config.js',
  ...params,
});
````